osu.Game/Beatmaps/BeatmapManager.cs

```diff
@@ -400,6 +411,82 @@ public void UndeleteAll()
 
         public Task Export(BeatmapSetInfo beatmap) => beatmapExporter.ExportAsync(beatmap.ToLive(Realm));
 
+        /// <summary>
+        /// Creates a copy of the <see cref="BeatmapSetInfo"/> and converts all beatmaps to legacy format, then exports it as a legacy package.
+        /// </summary>
+        /// <param name="beatmapSetInfo"></param>
+        /// <returns></returns>
+        public Task ExportLegacy(BeatmapSetInfo beatmapSetInfo)
+        {
+            // Create a clone of the original beatmap set which we will convert to legacy format and then export
+            var clone = new BeatmapSetInfo(beatmapSetInfo.Beatmaps.Select(b => b.Clone()));
+            clone.Files.AddRange(beatmapSetInfo.Files.Select(f => new RealmNamedFileUsage(f.File, f.Filename)));
+
+            // convert all beatmaps in the cloned beatmap set to legacy format
+            foreach (var beatmapInfo in clone.Beatmaps)
+            {
+                beatmapInfo.BeatmapSet = clone;
+                beatmapInfo.ID = Guid.NewGuid();
+
+                var file = beatmapInfo.File;
+                if (file == null)
+                    continue;
+
+                var beatmapContent = new LegacyBeatmapDecoder().Decode(new LineBufferedReader(RealmFileStore.Storage.GetStream(file.File.GetStoragePath())));
+                var beatmapSkin = new LegacySkin(new SkinInfo(), null!)
+                {
+                    Configuration = new LegacySkinDecoder().Decode(new LineBufferedReader(RealmFileStore.Storage.GetStream(file.File.GetStoragePath())))
+                };
+
+                Realm.Realm.Write(realm =>
+                {
+                    using var stream = new MemoryStream();
+                    convertAndEncodeLegacyBeatmap(beatmapContent, beatmapSkin, stream);
+
+                    beatmapInfo.MD5Hash = stream.ComputeMD5Hash();
+                    beatmapInfo.Hash = stream.ComputeSHA2Hash();
+
+                    file.File = RealmFileStore.Add(stream, realm).Detach();
+                });
+            }
+
+            return legacyBeatmapExporter.ExportAsync(new RealmLiveUnmanaged<BeatmapSetInfo>(clone));
+        }
+
+        private void convertAndEncodeLegacyBeatmap(IBeatmap beatmapContent, ISkin beatmapSkin, Stream stream)
```

Comment:
I'd also prefer it to be that way. The only issue is that a `LegacyExporter` does not have access to add files to the `RealmFileStore` and that would also be kinda weird. So without that I can't use the `LegacyArchiveExporter` as is. I'd have to modify it so it can do a conversion while getting the file streams. I'm pretty sure that's feasible though so I'll give it a try.

Reply:
I'd warn against overcomplicating it. If what it takes is to literally pass a `RealmFileStore` from beatmap manager to the exporter via an argument on `Export()`, so be it. No need for anything fancy.

Reply:
I've moved the logic to `LegacyBeatmapExporter` and I think its also considerably simpler now because it doesnt have to write to Realm